Sources/PackageModel/Manifest.swift

```diff
@@ -316,15 +322,25 @@ public struct TargetDescription: Equatable, Codable {
             case copy
         }
 
+        public enum Localization: String, Codable, Equatable {
+            case developmentRegion
+            case base
+        }
+
         /// The rule for the resource.
         public let rule: Rule
 
         /// The path of the resource.
         public let path: String
 
-        public init(rule: Rule, path: String) {
+        /// The explicit localization of the resource.
+        public let localization: Localization?
+
+        public init(rule: Rule, path: String, localization: Localization? = nil) {
+            precondition(rule == .process || localization == nil)
```

Comment:
Can we change the initializer to enforce this precondition?

Reply:
I can do:

```
        public init(copyPath path: String) {
            self.rule = .copy
            self.path = path
            self.localization = nil
        }

        public init(processPath path: String, localization: Localization? = nil) {
            self.rule = .process
            self.path = path
            self.localization = localization
        }
```

but it's a bit messy for callers which have a `Rule` instance and just want to pass it to the initializer.

Reply:
hmm, maybe this is fine. I would prefer if we could enforce this using the type system but it's probably not worth if it makes the usage messy